### PR TITLE
Updated the Dialog system to use a queue

### PR DESCRIPTION
**Pull request by VicFrank**
Dialog lines are added to a queue, instead of being owned by units. This means you can queue up a bunch of dialog at once, and it will properly bounce between units.

You can also clear all dialog in the queue (and stop the current dialog, including audio) by calling DialogController.clear();

### CODE REVIEW

File: content/panorama/layout/custom_game/custom_ui_manifest.xml [hunk 1]
```diff
@@ -9,10 +9,10 @@
         <!-- <include src="file://{resources}/scripts/custom_game/manifest.js" /> -->
     </scripts>
     <Panel>
+        <CustomUIElement type="Hud" layoutfile="file://{resources}/layout/custom_game/dialog.xml" />>
```
Comment (Perryvw) [inline content/panorama/layout/custom_game/custom_ui_manifest.xml:12]: double closing bracket here

File: game/scripts/vscripts/Dialog.ts [hunk 2]
```diff
@@ -144,37 +164,43 @@ class DialogController {
...
         gesture: GameActivity.DOTA_ATTACK,
         sendToAll: true,
         dialogStopsMovement: false,
-        sound: soundName
-    }])
+        sound: soundName,
+    });
 
-    dialogController.onDialogStart(hero, unit)
+    dialogController.onDialogStart(hero);
```
Comment (Perryvw) [inline game/scripts/vscripts/Dialog.ts:203]: Should onDialogStart be called here or when the dialog is popped from the queue?

**Comment by RobinKa** [comment]
So the dialog steps right now just wait a fixed amount of time. Is it possible to have a callback for when a dialog line ends for the step to detect?

**Comment by RobinKa** [comment]
Added the callback now. Skipping dialog and skipping between sections (so dialog gets stopped) works properly now. The dialog system seems pretty complicated though so I might have made a mistake somewhere.

**Pull request merged**